mapper/cumulocity/c8y_smartrest/src/smartrest_deserializer.rs

```diff
@@ -164,6 +164,73 @@ impl SmartRestUpdateSoftwareModule {
     }
 }
 
+fn to_datetime<'de, D>(deserializer: D) -> Result<DateTime<FixedOffset>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    // NOTE `NaiveDateTime` is used here because c8y uses for log requests a date time string which
+    // does not exactly equal `chrono::DateTime::parse_from_rfc3339`
+    // c8y result:
+    // 2021-10-23T19:03:26+0100
+    // rfc3339 expected:
+    // 2021-10-23T19:03:26+01:00
+    // so we add a ':'
+    let mut date_string: String = Deserialize::deserialize(deserializer)?;
+    let str_size = date_string.len();
+    date_string = date_string[0..str_size - 2].to_string()
+        + ":"
+        + &date_string[str_size - 2..str_size].to_string();
+    Ok(DateTime::parse_from_rfc3339(&date_string).unwrap())
+}
+
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
+pub struct SmartRestLogRequest {
+    pub message_id: String,
+    pub device: String,
+    pub log_type: String,
+    #[serde(deserialize_with = "to_datetime")]
+    pub date_from: DateTime<FixedOffset>,
+    #[serde(deserialize_with = "to_datetime")]
+    pub date_to: DateTime<FixedOffset>,
+    pub needle: Option<String>,
+    pub lines: usize,
+}
+
+impl SmartRestLogRequest {
+    pub fn new() -> Self {
+        Self {
+            message_id: "522".to_string(),
+            device: "".to_string(),
+            log_type: "".to_string(),
+            date_from: FixedOffset::east(0).ymd(1, 1, 1).and_hms(0, 0, 0),
+            date_to: FixedOffset::east(0).ymd(1, 1, 1).and_hms(0, 0, 0),
```

Comment:
@makr11st, @didier-wenzek let me know if you're happy with this for the default field. Previously if was `NaiveDateTime::from_timestamp(0, 0)`.